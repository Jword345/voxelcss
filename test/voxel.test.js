'use strict';

const ColorFace   = require('../lib/ColorFace');
const LightSource = require('../lib/LightSource');
const Mesh        = require('../lib/Mesh');
const Voxel       = require('../lib/Voxel');

describe('Voxel', function() {
    let voxel;
    beforeEach(() => {
        voxel = new Voxel();
    });
    it('can be instantiated with dimension 0', () => {
        expect(voxel.getDimension()).toEqual(0);
    });
    it('can create clones', function() {
        let x = 123;
        let y = 456;
        let z = 789;
        let original = new Voxel(x, y, z);
        let clone = original.clone();
        expect(clone.getPosition()).toMatchSnapshot();
    });
    it('can get and set mesh', () => {
        let oldmesh = voxel.getMesh();
        let newMesh = new Mesh(new ColorFace('#333'));
        expect(oldmesh).toMatchSnapshot();
        voxel.setMesh(newMesh);
        expect(voxel.getMesh()).toMatchSnapshot();
    });
    it('can update light sources', () => {
        let mesh = new Mesh(new ColorFace('#333'));
        voxel.updateLightSource(new LightSource());
        voxel.setMesh(mesh);
        expect(voxel.getMesh()).toMatchSnapshot();
    });
    it('can be added to and removed from a scene', function() {
        expect(() => {
            voxel.addToScene();
        }).toThrowErrorMatchingSnapshot();
        let scene = {
            add: jest.fn(),
            remove: jest.fn()
        };
        voxel.addToScene(scene);
        expect(scene.add).toHaveBeenCalledWith(voxel);
        voxel.removeFromScene();
        expect(scene.remove).toHaveBeenCalledWith(voxel);
        voxel.setParentScene();
        voxel.removeFromScene();
        expect(scene.remove).toHaveBeenCalledTimes(1);
    });
    it('can be added to a scene via animUp', function() {
        expect(() => {
            voxel.animUp();
        }).toThrowErrorMatchingSnapshot();
        let scene = {
            add: jest.fn(),
            remove: jest.fn()
        };
        voxel.addToScene(scene);
        expect(scene.add).toHaveBeenCalledWith(voxel);
    });
    it('can be added to a scene via animDown', function() {
        expect(() => {
            voxel.animUp();
        }).toThrowErrorMatchingSnapshot();
        let scene = {
            add: jest.fn(),
            remove: jest.fn()
        };
        voxel.addToScene(scene);
        expect(scene.add).toHaveBeenCalledWith(voxel);
    });
    it('can get and set dimensions', function() {
        let dim = 42;
        expect(voxel.getDimension()).toEqual(0);
        voxel.setDimension(dim);
        expect(voxel.getDimension()).toEqual(dim);
        voxel.setDimension('not a number');
        expect(voxel.getDimension()).toEqual(dim);
    });
});
