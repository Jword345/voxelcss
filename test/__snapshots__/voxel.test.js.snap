// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voxel can be added to a scene via animDown 1`] = `undefined`;

exports[`Voxel can be added to a scene via animUp 1`] = `undefined`;

exports[`Voxel can be added to and removed from a scene 1`] = `undefined`;

exports[`Voxel can create clones 1`] = `
Object {
  "x": 123,
  "y": 456,
  "z": 789,
}
`;

exports[`Voxel can get and set mesh 1`] = `
Mesh {
  "addEventListener": [Function],
  "getBack": [Function],
  "getBottom": [Function],
  "getFaces": [Function],
  "getFront": [Function],
  "getLeft": [Function],
  "getRight": [Function],
  "getTop": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setBack": [Function],
  "setBottom": [Function],
  "setFaces": [Function],
  "setFront": [Function],
  "setLeft": [Function],
  "setRight": [Function],
  "setTop": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Voxel can get and set mesh 2`] = `
Mesh {
  "addEventListener": [Function],
  "getBack": [Function],
  "getBottom": [Function],
  "getFaces": [Function],
  "getFront": [Function],
  "getLeft": [Function],
  "getRight": [Function],
  "getTop": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setBack": [Function],
  "setBottom": [Function],
  "setFaces": [Function],
  "setFront": [Function],
  "setLeft": [Function],
  "setRight": [Function],
  "setTop": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Voxel can update light sources 1`] = `
Mesh {
  "addEventListener": [Function],
  "getBack": [Function],
  "getBottom": [Function],
  "getFaces": [Function],
  "getFront": [Function],
  "getLeft": [Function],
  "getRight": [Function],
  "getTop": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setBack": [Function],
  "setBottom": [Function],
  "setFaces": [Function],
  "setFront": [Function],
  "setLeft": [Function],
  "setRight": [Function],
  "setTop": [Function],
  "triggerEvent": [Function],
}
`;
